src/main/java/edu/wpi/grip/ui/preview/PreviewsView.java

```diff
@@ -48,24 +59,57 @@ public PreviewsView(EventBus eventBus) {
     }
 
     @Subscribe
+    /**
+     * This function is called when a preview button is pushed/triggered
+     */
     public synchronized void onSocketPreviewChanged(SocketPreviewChangedEvent event) {
-        Platform.runLater(() -> {
```

Comment:
This is done to run the code in the UI thread.
Everything that manipulates the UI needs to run inside of the UI thread.